stores/sql/mysql/main.go

```diff
@@ -518,10 +522,45 @@ func (tx *MainDatabaseTx) RenameObjects(ctx context.Context, bucket, prefixOld,
 	return nil
 }
 
+func (tx MainDatabaseTx) SaveAccounts(ctx context.Context, accounts []api.Account) error {
+	// clean_shutdown = 1 after save
+	stmt, err := tx.Prepare(ctx, `
+		INSERT INTO ephemeral_accounts (created_at, account_id, clean_shutdown, host, balance, drift, requires_sync)
+		VAlUES (?, ?, 1, ?, ?, ?, ?)
+		ON DUPLICATE KEY UPDATE
+		account_id = VALUES(account_id),
+		clean_shutdown = 1,
+		host = VALUES(host),
+		balance = VALUES(balance),
+		drift = VALUES(drift),
+		requires_sync = VALUES(requires_sync)
+	`)
+	if err != nil {
+		return err
+	}
+	defer stmt.Close()
+
+	for _, acc := range accounts {
+		res, err := stmt.Exec(ctx, time.Now(), (ssql.PublicKey)(acc.ID), (ssql.PublicKey)(acc.HostKey), (*ssql.BigInt)(acc.Balance), (*ssql.BigInt)(acc.Drift), acc.RequiresSync)
+		if err != nil {
+			return fmt.Errorf("failed to insert account %v: %w", acc.ID, err)
+		} else if n, err := res.RowsAffected(); err != nil {
+			return fmt.Errorf("failed to get rows affected: %w", err)
+		} else if n != 1 && n != 2 { // 1 for insert, 2 for update
```

Comment:
how do you get two affected rows for an update?

Reply:
> With ON DUPLICATE KEY UPDATE, the affected-rows value per row is 1 if the row is inserted as a new row, 2 if an existing row is updated, and 0 if an existing row is set to its current values.

https://dev.mysql.com/doc/refman/5.7/en/insert-on-duplicate.html